### DISCUSSION TITLE
got it running under wine with newest version of winetricks

### DISCUSSION THREAD

**Discussion by shadow-absorber** [kwsch/PKHeX#3886] [General]
you can install 
`dotnetdesktop7`
using winetricks now which lets you run pkhex in a 64bit wine prefix
everything seems to work except for plugins which seems to cause stack overflows
but that is not nescarily a problem for kwsch to fix
here is a wine app db entry for one of the newer versions but not the newest version of pkhex:
[https://appdb.winehq.org/objectManager.php?sClass=version&iId=41479](https://appdb.winehq.org/objectManager.php?sClass=version&iId=41479)

**Comment by shadow-absorber** [comment]
these are the errors wine spits out if shadow tries to launch it with plugins in the plugins folder:
```
01ec:err:virtual:virtual_setup_exception stack overflow 2064 bytes addr 0x1700561ce stack 0x7a907f0 (0x7a90000-0x7a91000-0x7c10000)
01b4:err:virtual:virtual_setup_exception stack overflow 2000 bytes addr 0x170032480 stack 0x120830 (0x120000-0x121000-0x2a0000)
```

**Comment by shadow-absorber** [comment]
did not intend to close this

**Comment by teward** [comment]
FYI I already have pending version information for this submitted into the AppDB for the latest version.  Including that it needs updated winetricks to have `dotnetdesktop7`.  HOWEVER this will only work to get main PKHeX active, and it still has a crash tendency for things like EncounterDB, etc.

**Comment by teward** [comment]
For Awareness: I've adopted the application on the WineHQ AppDB as Super Maintainer on the AppDB, so I can add versions and test results without a long wait on the queue.

Requirements of updated winetricks and wine versions supporting `dotnetdesktop7` has been documented on the AppDB information since `(20)23.04.06` version which was accepted yesterday into the AppDB; for (20)23.05.11 the other god-maintainers told me to resubmit the `(20)23.05.11` version I had submitted directly as Super Maintainer so it bypasses the god-review queue, and I just did this a few minutes ago.

[WineHQ AppDB Entry for (20)23.04.06 version](https://appdb.winehq.org/objectManager.php?sClass=version&iId=41479) (Gold compatibility, Ubuntu 22.04, Wine 8.7)

[WineHQ AppDB Entry for (20)23.05.11 version](https://appdb.winehq.org/objectManager.php?sClass=version&iId=41511) (Reduction to Silver compatibility, Ubuntu 22.04, Wine 8.8; see details on the entry)

---

For @kwsch though: I think a link in the documentation to the [WineHQ AppDB entries for the PKHeX application](https://appdb.winehq.org/objectManager.php?sClass=application&iId=20317) may be sufficient enough to close this issue, since usually within a week or two I get to testing and publishing results on the AppDB including compatibility and setup steps, and issues being observed.

If not desired, then we're good to go without any action on the repository / coding / etc. side of things.  (and I'll just keep the AppDB updated)

**Comment by Lusamine** [comment]
Forgive me, but I'm not understanding why this needs to be linked on the main repo. Anyone can port it to any OS but it doesn't need to be endorsed on the main repository, nor does the readme need a constantly curated list of options for other OS's. Is your goal for visibility? In which case, this issue can be a discussion instead.

**Comment by teward** [comment]
@Lusamine I think OP is trying to say that a way to make this work from the .exe is with Wine on non-Windows distros, and is more a "Hey i have more info if you want to share it with users" rather than an actual support question.

This is reinforced by the fact that you cannot compile this even with dotnet7 on a non-Windows distro because the Windows Forms components are **Windows-specific** and are not in non-Windows dotnet7 versions.  Believe me, I tried compiling, it hard-fails on the Forms components.  Therefore, the only way to use this on non-Windows is with Wine.

(Note I intercepted this to indicate this is already documented on the WineHQ AppDB (where these kinds of "How to get working" bits are needed for version results and such) is all.)

**Comment by Lusamine** [comment]
Okay, I am converting this to a discussion then on how to get it to work on Wine as this is not a problem or goal of the program. When the program shifted to NET7, it was accepted that non-Windows support would become more limited as a result. Thus, it will be up to other people to figure out if they want it.

You can read this comment for more information: https://github.com/kwsch/PKHeX/issues/3623#issuecomment-1305068051
Note that there is a PPORG thread on that issue where people explain how to run it on Linux, so that is another avenue for discussion and information sharing.
